### ISSUE TITLE
dosbox-staging 0.75.1 prevents display idle sleep

### ISSUE THREAD

**Issue by Grounded0** [dosbox-staging/dosbox-staging#630]
**Describe the bug**
While dosbox-staging is running it prevents display sleep through SDL_DisableScreenSaver.

**To Reproduce**
Run dosbox-staging on macOS and enter terminal and type: `$pmset -g assertions`:

``` text
Assertion status system-wide:
...
   PreventUserIdleDisplaySleep    1
...
pid 83694(dosbox): [0x000e68bc0005a557]
00:00:07 PreventUserIdleDisplaySleep named: 
"dosbox-staging using SDL_DisableScreenSaver"
```

**Expected behaviour**
Emulator should not flag SDL_DisableScreenSaver as this prevents display from sleeping when using for example caffeinate on macOS which by default is configured to prevent system idle sleep while permitting display idle sleep.

**Environment (please complete the following information):**
macOS 10.15.6
2017 15" MacBook Pro

**Version of dosbox-staging:**
0.75.1

**Configuration parameters:**
dosbox-staging 0.75.1 off the website with dosbox-staging kernel and FreeDOS 1.2 userland with UNIVBE 5.3a loaded in AUTOEXEC.BAT.

**Additional context**
Probably impacts all platforms as it's in SDL 2.0: https://wiki.libsdl.org/SDL_DisableScreenSaver

**Comment by dreamer** [comment]
Some context: we do not turn off screensaver ourselves; according to SDL documentation, the screensaver is disabled by default since SDL 2.0.2 [source](https://wiki.libsdl.org/SDL_IsScreenSaverEnabled).

**Comment by dreamer** [comment]
@Grounded0 Thank you for the report :)

Test build for macOS will be available [here](https://github.com/dosbox-staging/dosbox-staging/runs/1173173020) in a moment - please check if it resolves the problem

I am not sure why SDL2 does this by default, but I guess there must be a reason (at least for some OSes) - I'll dig deeper to learn why they implemented it this way (my hunch is that some environment gets confused into thinking there's no user actions when mouse cursor is captured by SDL window, but who knows… I would prefer to avoid adding new .conf setting if possible but perhaps we will need to do it.

**Comment by grapeli** [comment]
You can always change this behavior.
https://wiki.libsdl.org/FAQUsingSDL
`SDL_VIDEO_ALLOW_SCREENSAVER=1 dosbox-staging`
or
`export SDL_VIDEO_ALLOW_SCREENSAVER=1`

**Comment by kcgen** [comment]
Nice find @grapeli !
Assuming this code-change does what we expect, then the flip scenario: `SDL_VIDEO_ALLOW_SCREENSAVER=0` might be useful in the rare case where users actually want dosbox to override and block their hosts's screensaver/sleep settings. 

(Curious if anyone eventually requests this)

**Comment by dreamer** [comment]
@grapeli @kcgen what do you think of creating a new .conf setting for this? yes/no/don't care?

Because if we'll add note about `SDL_VIDEO_ALLOW_SCREENSAVER` to our README file, then in practice rarely anyone will use it.

But if we'll create:
```
[sdl]
allow_screensaver = true
```
Then it'll be easier for users to discover it.

I imagine some users with CRTs might want to suppress OS screensaver, so screensaver built-in their DOS software (Norton Commander, or e.g. Windows 3.11) would take the precedence.

**Comment by kcgen** [comment]
I would opt for no conf settings - and permitting the host to behave per the user's wishes. 

Museum PCs (say, on for 8hrs a day displaying historic software), or users wanting an 'always on' DOSBox system are likely to disable sleeping at the host level. 

Also  foing by the fact that 'not blocking the screensaver' was default in SDL up to 2.0.2 yet there were no reports on Vogons (in 15+ years) where users called for dosbox to block their configured host's configured idle settings.

**Comment by dreamer** [comment]
Wait, I don't get it - if SDL1 was not blocking the screensaver, then why people were complaining on Vogons? Maybe SDL1 *was* blocking it (hence complains), SDL 2.0.0 and SDL 2.0.1 wasn't, and then SDL team brought the old behaviour back?

**Comment by kcgen** [comment]
Yes; I'm not sure if pre-2.0.2 SDL actively tried to allow or block the host's idle-detection routines; I know this has been a bit of a cat and mouse game atleast on Android and Windows vs their apps.  

Understood regarding allowing the ENV variable to override the conf setting.  But my vote still stands.. there are zero requests of users wanting dosbox to override and block their configured screensaver settings; so I feel that this would be an unnecessary conf option.

**Comment by Grounded0** [comment]
I actually found a decent way to override per application settings in macOS. When you enter lock screen press the Cancel button at the bottom of the lock screen. This will override any per application settings and the system will forcibly enter into requested sleep mode, in my case idle display sleep as set in caffeinate.

**Comment by kcgen** [comment]
> When you enter lock screen press the Cancel button at the bottom to override any per application settings and forcibly enter into requested sleep mode

Nice to see MacOS has a sledge hammer to knock out any blockers! 🔨   With countless apps being a thin wrapper around user data harvesting for 'affiliates' and ad dissemination, there's unfortunately a financial motive to block sleep and keep the party going; only for users to wake up and find their battery drained.

**Comment by dreamer** [comment]
@Grounded0 glad there's a workaround :) do you have a moment to test [this build](https://github.com/dosbox-staging/dosbox-staging/actions/runs/275369889) and see if workaround is needed with it?

**Comment by Grounded0** [comment]
@dreamer 

ajp@milla:~$ pmset -g assertions
2020-09-27 23:11:09 +0300 
Assertion status system-wide:
   BackgroundTask                 0
   ApplePushServiceTask           0
   UserIsActive                   1
   PreventUserIdleDisplaySleep    1
   PreventSystemSleep             0
   ExternalMedia                  1
   InternalPreventDisplaySleep    1
   PreventUserIdleSystemSleep     1
   NetworkClientActive            0
Listed by owning process:
   pid 210(coreaudiod): [0x000f22e300018f76] 00:02:24 PreventUserIdleSystemSleep named: "com.apple.audio.AppleHDAEngineOutput:1F,3,0,1,1:0.context.preventuseridlesleep"  
        Created for PID: 88006. 
   pid 108(powerd): [0x000f16d100088dd3] 00:53:54 ExternalMedia named: "com.apple.powermanagement.externalmediamounted"  
   pid 108(powerd): [0x000f20af00108001] 00:02:35 InternalPreventDisplaySleep named: "com.apple.powermanagement.delayDisplayOff"  
        Timeout will fire in 144 secs Action=TimeoutActionTurnOff
   pid 903(UserEventAgent): [0x000f1ac8000b8e51] 00:36:59 BackgroundTask named: "com.apple.photoanalysisd.backgroundanalysis"  
        Created for PID: 4027. 
   pid 2613(caffeinate): [0x0000149600018486] 277:06:12 PreventUserIdleSystemSleep named: "caffeinate command-line tool"  
        Details: caffeinate asserting forever
        Localized=THE CAFFEINATE TOOL IS PREVENTING SLEEP.
   pid 88006(dosbox): [0x000f22e100058f6f] 00:02:26 PreventUserIdleDisplaySleep named: "dosbox-staging using SDL_DisableScreenSaver"  
   pid 151(hidd): [0x000f1d0900098e9c] 00:00:00 UserIsActive named: "com.apple.iohideventsystem.queue.tickle serviceID:10000039e name:AppleHIDKeyboardEve product:Apple Internal Keyb eventType:3"  
        Timeout will fire in 300 secs Action=TimeoutActionRelease
Kernel Assertions: 0x104=USB,MAGICWAKE
   id=504  level=255 0x100=MAGICWAKE mod=23.09.2020, 10.34 description=llw0 owner=IOSkywalkNetworkBSDClient
   id=505  level=255 0x100=MAGICWAKE mod=16.09.2020, 14.07 description=en0 owner=en0
   id=513  level=255 0x4=USB mod=01.01.1970, 2.00 description=com.apple.usb.externaldevice.01100000 owner=Ultra USB 3.0
Idle sleep preventers: IODisplayWrangler

Executable being run while 0.75.1 quitted: dosbox-staging-macOS-v0.76.0-alpha-754-g45d3d.dmg

**Comment by kcgen** [comment]
Interesting - the coraudio daemon is, itself, preventing user idle sleep because it's busy on behalf of dosbox-staging.

``` text
210(coreaudiod): [0x000f22e300018f76]
00:02:24 PreventUserIdleSystemSleep named:
"com.apple.audio.AppleHDAEngineOutput:1F,3,0,1,1:0.context.preventuseridlesleep"
Created for PID: 88006.
```

However, strange that it says dosbox is still block due to `SDL_DisableScreenSaver`.

What happens if you launch it with @grapeli's suggestion: `SDL_VIDEO_ALLOW_SCREENSAVER=1 dosbox-staging`?

**Comment by Grounded0** [comment]
Do i have to pass it as arguments to the executable or what?

**Comment by kcgen** [comment]
If you open a terminal, find the dosbox executable, and just launch with the entire line:

`SDL_VIDEO_ALLOW_SCREENSAVER=1 ./dosbox`

Or you can set as an environment variable and then launch:

1. `export SDL_VIDEO_ALLOW_SCREENSAVER=1`
1. `./dosbox`

**Comment by Grounded0** [comment]
Actually i did some detective work. SheepShaver also passes such argument to Coreaudio but because Coreaudio is an Apple API it is smart enough to release such condition if user requests non hibernating sleep modes through screen lock.

**Comment by kcgen** [comment]
Do you have more detail? Coreaudio itself is not an SDL application, so SDL_VIDEO_ALLOW_SCREENSAVER=1 will (likely) have no effect on it.

Does sheepshaver pass coreaudio something else to allow it to sleep?

**Comment by Grounded0** [comment]
ajp@milla:/Volumes/dosbox-staging$ open dosbox-staging.app --args SDL_VIDEO_ALLOW_SCREENSAVER=1

(this is way we pass arguments to .APP application packages)

ajp@milla:~$ pmset -g assertions
2020-09-28 00:38:23 +0300 
Assertion status system-wide:
   BackgroundTask                 0
   ApplePushServiceTask           0
   UserIsActive                   1
   PreventUserIdleDisplaySleep    1
   PreventSystemSleep             0
   ExternalMedia                  0
   InternalPreventDisplaySleep    1
   PreventUserIdleSystemSleep     1
   NetworkClientActive            0
Listed by owning process:
   pid 210(coreaudiod): [0x000f37d60001932e] 00:00:13 PreventUserIdleSystemSleep named: "com.apple.audio.AppleHDAEngineOutput:1F,3,0,1,1:0.context.preventuseridlesleep"  
        Created for PID: 88426. 
   pid 903(UserEventAgent): [0x000f1ac8000b8e51] 02:04:12 BackgroundTask named: "com.apple.photoanalysisd.backgroundanalysis"  
        Created for PID: 4027. 
   pid 108(powerd): [0x000f375f00108001] 00:02:13 InternalPreventDisplaySleep named: "com.apple.powermanagement.delayDisplayOff"  
        Timeout will fire in 167 secs Action=TimeoutActionTurnOff
   pid 2613(caffeinate): [0x0000149600018486] 278:33:26 PreventUserIdleSystemSleep named: "caffeinate command-line tool"  
        Details: caffeinate asserting forever
        Localized=THE CAFFEINATE TOOL IS PREVENTING SLEEP.
   pid 88426(dosbox): [0x000f37d400059327] 00:00:16 PreventUserIdleDisplaySleep named: "dosbox-staging using SDL_DisableScreenSaver"  
   pid 151(hidd): [0x000f1d0900098e9c] 00:00:00 UserIsActive named: "com.apple.iohideventsystem.queue.tickle serviceID:10000039e name:AppleHIDKeyboardEve product:Apple Internal Keyb eventType:3"  
        Timeout will fire in 300 secs Action=TimeoutActionRelease
Kernel Assertions: 0x100=MAGICWAKE
   id=504  level=255 0x100=MAGICWAKE mod=23.09.2020, 10.34 description=llw0 owner=IOSkywalkNetworkBSDClient
   id=505  level=255 0x100=MAGICWAKE mod=16.09.2020, 14.07 description=en0 owner=en0
Idle sleep preventers: IODisplayWrangler

**Comment by kcgen** [comment]
It's not an argument.
It needs to be part of the environment. 
In your shell, can you run:

``` shell
$ export SDL_VIDEO_ALLOW_SCREENSAVER=1
$ open dosbox-staging.app
```

**Comment by Grounded0** [comment]
SheepShaver running (correct behaviour):

ajp@milla:~$ pmset -g assertions
2020-09-28 00:42:08 +0300 
Assertion status system-wide:
   BackgroundTask                 0
   ApplePushServiceTask           0
   UserIsActive                   1
   PreventUserIdleDisplaySleep    0
   PreventSystemSleep             0
   ExternalMedia                  0
   InternalPreventDisplaySleep    1
   PreventUserIdleSystemSleep     1
   NetworkClientActive            0
Listed by owning process:
   pid 210(coreaudiod): [0x000f38aa0001934a] 00:00:28 PreventUserIdleSystemSleep named: "com.apple.audio.AppleHDAEngineOutput:1F,3,0,1,1:0.context.preventuseridlesleep"  
        Created for PID: 88436. 
   pid 903(UserEventAgent): [0x000f1ac8000b8e51] 02:07:58 BackgroundTask named: "com.apple.photoanalysisd.backgroundanalysis"  
        Created for PID: 4027. 
   pid 108(powerd): [0x000f38a400108001] 00:00:33 InternalPreventDisplaySleep named: "com.apple.powermanagement.delayDisplayOff"  
        Timeout will fire in 267 secs Action=TimeoutActionTurnOff
   pid 2613(caffeinate): [0x0000149600018486] 278:37:11 PreventUserIdleSystemSleep named: "caffeinate command-line tool"  
        Details: caffeinate asserting forever
        Localized=THE CAFFEINATE TOOL IS PREVENTING SLEEP.
   pid 151(hidd): [0x000f1d0900098e9c] 00:00:00 UserIsActive named: "com.apple.iohideventsystem.queue.tickle serviceID:10000039e name:AppleHIDKeyboardEve product:Apple Internal Keyb eventType:3"  
        Timeout will fire in 300 secs Action=TimeoutActionRelease
Kernel Assertions: 0x100=MAGICWAKE
   id=504  level=255 0x100=MAGICWAKE mod=23.09.2020, 10.34 description=llw0 owner=IOSkywalkNetworkBSDClient
   id=505  level=255 0x100=MAGICWAKE mod=16.09.2020, 14.07 description=en0 owner=en0
Idle sleep preventers: IODisplayWrangler

As you can see    PreventUserIdleDisplaySleep    0 is zero.

Project source code is here:
https://github.com/cebix/macemu

I think i'll comb through it to see any clues.

**Comment by Grounded0** [comment]
ajp@milla:~$ export SDL_VIDEO_ALLOW_SCREENSAVER=1

ajp@milla:~/Downloads/unprocessed$ open dosbox-staging.app   

ajp@milla:~/Downloads/unprocessed$ pmset -g assertions           
2020-09-28 00:49:17 +0300                                                                                                  
Assertion status system-wide:                                
   BackgroundTask                 0                       
   ApplePushServiceTask           0                                                                                        
   UserIsActive                   1                          
   PreventUserIdleDisplaySleep    0                                                                                        
   PreventSystemSleep             0                                                                                        
   ExternalMedia                  0                                                                                        
   InternalPreventDisplaySleep    1
   PreventUserIdleSystemSleep     1                                                                                        
   NetworkClientActive            0                                                                                        
Listed by owning process:                                    
   pid 210(coreaudiod): [0x000f3a550001938b] 00:00:29 PreventUserIdleSystemSleep named: "com.apple.audio.AppleHDAEngineOutput:1F,3,0,1,1:0.context.preventuseridlesleep"  
        Created for PID: 88510.          
   pid 903(UserEventAgent): [0x000f1ac8000b8e51] 02:15:06 BackgroundTask named: "com.apple.photoanalysisd.backgroundanalysis"                                                                                                                         
        Created for PID: 4027.                                                                                             
   pid 73415(mds_stores): [0x000f3a72000b938e] 00:00:00 BackgroundTask named: "com.apple.metadata.mds_stores.power"        
   pid 2613(caffeinate): [0x0000149600018486] 278:44:20 PreventUserIdleSystemSleep named: "caffeinate command-line tool"   
        Details: caffeinate asserting forever                
        Localized=THE CAFFEINATE TOOL IS PREVENTING SLEEP.   
   pid 108(powerd): [0x000f38a400108001] 00:00:57 InternalPreventDisplaySleep named: "com.apple.powermanagement.delayDisplayOff"             
        Timeout will fire in 243 secs Action=TimeoutActionTurnOff                                                          
   pid 151(hidd): [0x000f1d0900098e9c] 00:00:00 UserIsActive named: "com.apple.iohideventsystem.queue.tickle serviceID:10000039e name:AppleHIDKeyboardEve product:Apple Internal Keyb eventType:3"                                                    
        Timeout will fire in 300 secs Action=TimeoutActionRelease                                                          
Kernel Assertions: 0x100=MAGICWAKE               
   id=504  level=255 0x100=MAGICWAKE mod=23.09.2020, 10.34 description=llw0 owner=IOSkywalkNetworkBSDClient                
   id=505  level=255 0x100=MAGICWAKE mod=16.09.2020, 14.07 description=en0 owner=en0                                       
Idle sleep preventers: IODisplayWrangler

Seems to be working and after actual testing seems to be working as intended.

**Comment by kcgen** [comment]
That's great news @Grounded0.

@dreamer ; could it be that the environment variable has a different effect on SDL versus the internal API call?

Or perhaps (which I suspect) some subsequent initialization step is silently resetting this state (at which point, it re-queries the env variable).

**Comment by dreamer** [comment]
@Grounded0 New build to test will be here in a moment: [link](https://github.com/dosbox-staging/dosbox-staging/actions/runs/275617514) - test if *without* setting the environment variable.

@kcgen I was setting the hint early, before any window is being created, but maybe it's too late already? Not sure.

I replaced `SDL_SetHint` call suggested in SDL wiki (with flag introduced in 2.0.2) with `SDL_EnableScreensaver` (which is available since 2.0.0). Let's see if it changes anything.

**Comment by Grounded0** [comment]
I also noticed that Coreaudio is set as PreventUserIdleSystemSleep NOT PreventUserIdleDisplaySleep. These are two different sleep states and are not part of the problem we're dealing with.

**Comment by Grounded0** [comment]
Arguments cleared (dosbox-staging-macOS-v0.76.0-alpha-754-g45d3d behaves badly).

ajp@milla:~/Downloads/unprocessed$ open dosbox-staging-macOS-v0.76.0-alpha-755-g5530b/dosbox-staging.app

ajp@milla:~/Downloads/unprocessed/dosbox-staging-macOS-v0.76.0-alpha-755-g5530b$ pmset -g assertions
2020-09-28 01:18:23 +0300 
Assertion status system-wide:
   BackgroundTask                 0
   ApplePushServiceTask           0
   UserIsActive                   1
   PreventUserIdleDisplaySleep    0
   PreventSystemSleep             0
   ExternalMedia                  0
   InternalPreventDisplaySleep    1
   PreventUserIdleSystemSleep     1
   NetworkClientActive            0
Listed by owning process:
   pid 210(coreaudiod): [0x000f412e00019496] 00:00:22 PreventUserIdleSystemSleep named: "com.apple.audio.AppleHDAEngineOutput:1F,3,0,1,1:0.context.preventuseridlesleep"  
        Created for PID: 88682. 
   pid 903(UserEventAgent): [0x000f1ac8000b8e51] 02:44:13 BackgroundTask named: "com.apple.photoanalysisd.backgroundanalysis"  
        Created for PID: 4027. 
   pid 108(powerd): [0x000f3e7e00108001] 00:00:24 InternalPreventDisplaySleep named: "com.apple.powermanagement.delayDisplayOff"  
        Timeout will fire in 276 secs Action=TimeoutActionTurnOff
   pid 2613(caffeinate): [0x0000149600018486] 279:13:26 PreventUserIdleSystemSleep named: "caffeinate command-line tool"  
        Details: caffeinate asserting forever
        Localized=THE CAFFEINATE TOOL IS PREVENTING SLEEP.
   pid 151(hidd): [0x000f1d0900098e9c] 00:00:00 UserIsActive named: "com.apple.iohideventsystem.queue.tickle serviceID:10000039e name:AppleHIDKeyboardEve product:Apple Internal Keyb eventType:3"  
        Timeout will fire in 300 secs Action=TimeoutActionRelease
Kernel Assertions: 0x100=MAGICWAKE
   id=504  level=255 0x100=MAGICWAKE mod=23.09.2020, 10.34 description=llw0 owner=IOSkywalkNetworkBSDClient
   id=505  level=255 0x100=MAGICWAKE mod=16.09.2020, 14.07 description=en0 owner=en0
Idle sleep preventers: IODisplayWrangler

Looking as it should be and passed actual testing three times.

**Comment by dreamer** [comment]
I tested on Linux / Gnome 3.36.6.

The exact bug cannot be replicated (when user locks the machine, Gnome overrides any application that might want to keep the display on), but when leaving the desktop running without any input dosbox-staging did indeed suppress blanking of the screen.

After this change, dosbox-staging does not do this any more - OS fades out / disables the screen the same it does for other apps :)

Never noticed this issue before because my screensaver is set up to start after a long time.

In discussions on Vogons around this issue, someone said, that upstream wants screensaver suppressed to avoid problems when using custom resolution after screensaver changes it to OS one. This was a valid concern for early 2000s when many users still had CRTs, but nowadays - not so much (dosbox-staging defaults to `fullresolution=desktop` anyway). Just to be thorough I tested this situation anyway - on leaving screensaver Gnome *did* change my resolution from custom one back to FHD… but dosbox-staging handled the case gracefully (thanks, SDL2!) and switched to the same mode as if `fullresolution=desktop` was used :).

**Comment by Grounded0** [comment]
I think someone should observe behaviour on Windows too so we get this right.

**Comment by kcgen** [comment]
> edit The one use-case I'm still worried about is watching long videos or cutscenes in the games… @kcgen what are your thoughts on this?

Right; with a very short screensaver/idle-mode timeout (say between 60s and a couple minutes), I can definitely see it activating if the user sits and watches without generating IO input.

Suggest we see if users complain about this though. 

For these hyper-power sensitive users, they might already be used this behavior.. ie, just reading a big wiki page might bring up the screensaver, so they probably have tools or muscle memory to keep things alive.  

For these users, my guess is the guaranteed power-saving is a desired priority over dosbox accidentally draining their system. (For example, thinking of a nomad van user running off a small solar panel and battery; every watt collected counts!).

**Comment by dreamer** [comment]
I repeated tests on Windows 10, hoping it will be a quick ordeal, but nope :(

1. When Windows has screen blanking set to kick in after 1 minute (the lowest possible setting), it actually starts after 45s - 55s. And it does not gently fade the screen to black the way Gnome does - it just abruptly kicks in. This is a Windows problem (not our concern, but I wanted to document it so other testers won't be surprised).
2.  Our current implementation does not suppress the screensaver when started with `SDL_VIDEO_ALLOW_SCREENSAVER=0` (not a big problem, this would require only a tiny fix in #637.
3. When testing (2), I placed dosbox-staging window on my left screen… **and it broke my display**. Windows was unable to recover correctly - my monitor went through several "broken" states, then recovered partially. Content of the screen was ok, but the whole display seemed to have green channel partially disabled (everything had magenta hue). I was afraid it actually physically broke my monitor, but thankfully it went back to normal after restarting my PC.

Went back to Fedora to repeat tests. (2) is reproducible; (3) is **not** reproducible on Linux. Gnome always recovers correctly without any problem (no matter what display or environment variables I use).

What's notable about (3) - it happened only when I placed dosbox-staging fullscreen on my left display. When I was testing my main display (right), Windows was able to recover correctly. I don't know if it's a bug in Windows, or in Windows's shitty drivers, or maybe even in SDL2 (which I find very unlikely). I was using `fullresolution=desktop` and `output=opengl` (our defaults). We can't rule out that it's a bug in drivers and might affect other OSes as well (especially macOS - because OpenGL is deprecated in there).

@kcgen At this point, I think we should keep existing behaviour (suppress the screensaver by default), introduce a setting for users to override it *and place a warning in the documentation for this setting*.

**Comment by kcgen** [comment]
> @kcgen At this point, I think we should keep existing behaviour (suppress the screensaver by default), introduce a setting for users to override it and place a warning in the documentation for this setting.

Wow - scary, and great you were able to test on real Windows hardware! Given this, your alternative looks perfect.

**Comment by dreamer** [comment]
@Grounded0 This issue got closed as we merged new fix for this problem.

We can't allow screensaver to work automatically (reasons are explained in previous comments and in #717), but there's a new option available now:

```
[sdl]
screensaver = allow
```
When using it, I **strongly** recommend keeping `fullresolution = desktop` (especially if you're using more than 1 display).

This new option will be released in upcoming 0.76.0 version :)

**Comment by vmedea** [comment]
> edit The one use-case I'm still worried about is watching long videos or cutscenes in the games… @kcgen what are your thoughts on this?

Also when using a joystick/gamepad this setting is very welcome. At least on this particular Linux setup, it doesn't seem like joystick activity is seen as user activity with regard to the screensaver.
